### PR TITLE
Add reporter implementation project

### PR DESCRIPTION
**Pull request by cdelg**
See issue #2138

Signed-off-by: Clément Delgrange <cl.delgrange@protonmail.com>

### CODE REVIEW

**Comment by cdelg** [comment]
The build failed but I don't understand why. Any idea?

**Comment by bjhargrave** [comment]
> The build failed but I don't understand why. Any idea?

No. But it is a meaningless failure. For some reason, sometimes on appveyor, the build is successful, but it then complains at the end that a windows batch file is missing...

File: biz.aQute.bnd.reporter/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	org.jtwig:jtwig-reflection;version=5.86,\
+	com.googlecode.concurrentlinkedhashmap.lru;version=1.4,\
+	org.objectweb.asm;version=5.0,\
+	org.objectweb.asm.analysis;version=5.0,\
+	org.objectweb.asm.tree;version=5.0,\
+	org.objectweb.asm.util;version=5.0,\
+	org.parboiled:parboiled-core;version=1.1
+	
+-testpath: \
+	org.apache.servicemix.bundles.junit
```
Comment (bjhargrave) [inline biz.aQute.bnd.reporter/bnd.bnd]: This should be the `${junit}` macro rather than a specific junit bundle.

**Comment by cdelg** [comment]
> I'll need to set aside some time to review/build/play with this. One that that is missing is some documentation on how I would use it. Is there some command line I would invoke?

There is no command line in this PR. The bnd command line was in the impl-demo branch, unfortunately I removed this branch :(
I can create a new branch with the command line implementation, or I can send you the jar directly if you prefer. Then, you will be able to configure your report in your projects and use this command line to generate them. The command line also allow to create reports for external jar. 

> Or do I configure it in my build? For example after building this and releasing it, could the bnd build itself use this and how?

There is no integration with the build process (if this is what you mean), before this current version @pkriens created a postBuild plugin for that, but finally the main use case is to generate these reports occasionally via a maven/gradle task or the command line. This feature could be done later. 

Sometimes ago I setup a demo workspace I could update it and send you the link.

**Comment by cdelg** [comment]
I have fixed some bugs but I squash the commit :(

I have a small repo at [https://github.com/cdelg/org.cdecode.docdemo](https://github.com/cdelg/org.cdecode.docdemo).

The compiled bnd command line is at the root of the workspace and you can run it on this repo (or others). There is template files also but they are not up to date so the resulting readme file are not perfect (only the main readme is good). But you can start to use the command line to generate .json  or .xml file without using the templates (or create your owns).

I will provide more documentation, and a better demo project.

**Comment by cdelg** [comment]
The report configurations are at the end of the build.bnd file.

**Comment by cdelg** [comment]
I have updated the repository at https://github.com/cdelg/org.cdecode.docdemo. Now all the templates work and can be used to test the feature. A documentation is available in this repo [https://github.com/cdelg/org.cdecode.docdemo/blob/master/documentation.md](https://github.com/cdelg/org.cdecode.docdemo/blob/master/documentation.md).

**Comment by cdelg** [comment]
Also if you can retry the build...

**Comment by cdelg** [comment]
The repository project tests failed but I don't know why, also it seems that the reporter project is not compiled/tested by the build, is there something I can do?

**Comment by cdelg** [comment]
What is the usage of the BndPlugin.class annotation?

**Comment by bjhargrave** [comment]
Hi, we just released 4.1. Master is now at 4.2. Can you rebase on master? Then we can look to merge this in early in the 4.2 release cycle.

**Comment by cdelg** [comment]
done

**Comment by bjhargrave** [comment]
You need to amend your commits so that each commit message includes a `Signed-off-by` as required by CONTRIBUTING. Then push force your updated commits. Thanks.

**Comment by bjhargrave** [comment]
You need to amend your commits so that each commit message includes a `Signed-off-by` as required by CONTRIBUTING. Then push force your updated commits. Thanks.

**Comment by cdelg** [comment]
I pushed the commits, but they are not reflected here, I think that it is related to the current status of Github.

**Comment by cdelg** [comment]
I pushed the commits, but they are not reflected here, I think that it is related to the current status of Github.

**Comment by bjhargrave** [comment]
You need to amend your commits so that each commit message includes a `Signed-off-by` as required by CONTRIBUTING. Then push force your updated commits. Thanks.

**Comment by cdelg** [comment]
I pushed the commits, but they are not reflected here, I think that it is related to the current status of Github.

**Comment by cdelg** [comment]
I pushed the commits, but they are not reflected here, I think that it is related to the current status of Github.

**Comment by cdelg** [comment]
I pushed the commits, but they are not reflected here, I think that it is related to the current status of Github.

**Comment by cdelg** [comment]
I pushed the commits, but they are not reflected here, I think that it is related to the current status of Github.

**Comment by cdelg** [comment]
I pushed the commits, maybe you will need to trigger the build due to GitHub error status. Changes are not reflected in the PR yet.

**Comment by cdelg** [comment]
Do I need to squash the commits?

**Comment by bjhargrave** [comment]
> Do I need to squash the commits?

Lets do that. 

I checkout the PR locally and built it. The project is not part of the main build since it is not referenced anywhere in `dist/bnd.bnd` or by any other project. If you want it part of the build, then you will need to add a commit to reference it in the `-dependson` in `dist/bnd.bnd`.

**Comment by cdelg** [comment]
> If you want it part of the build, then you will need to add a commit to reference it in the -dependson in dist/bnd.bnd

Maybe you can tell me if you want this to be part of the build, because I don't know :/ ? (There is another and last PR which will add this feature to the bnd command line)

So, if you want I can add the `-dependson`, squash and push?

**Comment by bjhargrave** [comment]
We can defer the add to `dist/bnd.bnd` but until the project is referenced in the build, it is just files in the git repo and is not built in the CI builds.

**Comment by cdelg** [comment]
done

**Pull request merged**